### PR TITLE
VS Navigation consolidation

### PR DESCRIPTION
**Pull request by TIHan**
We have several ways to do navigation in VS, but a lot of it was spread out. This PR aims to consolidate it into a single type `FSharpNavigation`. 

We have some quirks between quick-info and go-to-definition such as if one doesn't work the other one will? The goal was to address that, but it turns out it's going to be a bit more involved because the different navigation calls are catered to their respective VS feature. We will need to address that part in a separate PR.

### CODE REVIEW

**Comment by dsyme** [comment]
This is great!

File: vsintegration/src/FSharp.Editor/Navigation/GoToDefinition.fs [hunk 1]
```diff
@@ -488,4 +491,237 @@ type internal GoToDefinition(metadataAsSource: FSharpMetadataAsSourceService) =
...
+        // Wrap this in a try/with as if the user clicks "Cancel" on the thread dialog, we'll be cancelled.
+        // Task.Wait throws an exception if the task is cancelled, so be sure to catch it.
+        try
+            // This call to Wait() is fine because we want to be able to provide the error message in the status bar.
+            gtdTask.Wait(cancellationToken)
+            if gtdTask.Status = TaskStatus.RanToCompletion && gtdTask.Result.IsSome then
+                match gtdTask.Result.Value with
+                | FSharpGoToDefinitionResult.NavigableItem(navItem), _ ->
+                    gtd.NavigateToItem(navItem, statusBar)
+                    // 'true' means do it, like Sheev Palpatine would want us to.
```
Comment (cartermp) [inline vsintegration/src/FSharp.Editor/Navigation/GoToDefinition.fs:713]: do it

File: vsintegration/src/FSharp.Editor/Navigation/GoToDefinition.fs [hunk 2]
```diff
@@ -488,4 +491,237 @@ type internal GoToDefinition(metadataAsSource: FSharpMetadataAsSourceService) =
...
+                    true
+                | FSharpGoToDefinitionResult.ExternalAssembly(targetSymbolUse, metadataReferences), _ ->
+                    gtd.NavigateToExternalDeclaration(targetSymbolUse, metadataReferences, cancellationToken, statusBar)
+                    // 'true' means do it, like Sheev Palpatine would want us to.
```
Comment (cartermp) [inline vsintegration/src/FSharp.Editor/Navigation/GoToDefinition.fs:717]: do it

**Review by cartermp — COMMENTED** [review]
More a nit about calling this navigation. We also have a `FSharpNavigateToSearchService` that is technically a form of navigation, but has nothing to do with navigating to the definition of a symbol.

**Comment by TIHan** [comment]
This is ready

**Review by dsyme — APPROVED** [review]

**Comment by dsyme** [comment]
Great work!

**Pull request merged**